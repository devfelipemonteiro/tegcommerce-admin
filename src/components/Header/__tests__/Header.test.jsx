import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';
import MenuItem from '../../MenuItem';

const eventMockObject = {
  preventDefault: jest.fn()
};

const minimalProps = {
  pushFn: jest.fn(),
  openSidebarFn: jest.fn()
};

describe('<Header /> Tests', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render without exploding', done => {
    const wrapper = shallow(<Header {...minimalProps} />);
    expect(wrapper).toHaveLength(1);
    done();
  });

  it('should render correctly with minimal props', done => {
    const wrapper = shallow(<Header {...minimalProps} />);
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('should call pushFn when click on Home menu item', done => {
    const wrapper = shallow(<Header {...minimalProps} />);
    const menuItem = wrapper.find(MenuItem).filterWhere(item => item.children().contains('Home'));
    const menuItemAnchor = menuItem
      .dive()
      .shallow()
      .find('a')
      .first();
    menuItemAnchor.simulate('click', eventMockObject);
    expect(minimalProps.pushFn.mock.calls.length).toBe(1);
    done();
  });

  it('should call pushFn when click on Sign In menu item', done => {
    const wrapper = shallow(<Header {...minimalProps} />);
    const menuItem = wrapper
      .find(MenuItem)
      .filterWhere(item => item.children().contains('Sign In'));
    const menuItemAnchor = menuItem
      .dive()
      .shallow()
      .find('a')
      .first();
    menuItemAnchor.simulate('click', eventMockObject);
    expect(minimalProps.pushFn.mock.calls.length).toBe(1);
    done();
  });

  it('should call pushFn when click on Sign Up menu item', done => {
    const wrapper = shallow(<Header {...minimalProps} />);
    const menuItem = wrapper
      .find(MenuItem)
      .filterWhere(item => item.children().contains('Sign Up'));
    const menuItemAnchor = menuItem
      .dive()
      .shallow()
      .find('a')
      .first();
    menuItemAnchor.simulate('click', eventMockObject);
    expect(minimalProps.pushFn.mock.calls.length).toBe(1);
    done();
  });
});
