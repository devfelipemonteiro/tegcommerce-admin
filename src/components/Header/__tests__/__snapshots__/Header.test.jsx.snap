// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Tests should render correctly with minimal props 1`] = `
<header
  className="root"
>
  <div
    className="container"
  >
    <div
      className="contentLeftContainer"
    >
      <Menu
        style={
          Object {
            "root": "menu__root",
          }
        }
      >
        <MenuItem
          href="/"
          onClick={[Function]}
          style={
            Object {
              "container": "menuItem__container",
            }
          }
        >
          Home
        </MenuItem>
      </Menu>
    </div>
    <div
      className="contentRightContainer"
    >
      <MenuItem
        href="signin"
        onClick={[Function]}
        style={
          Object {
            "container": "menuItem__container",
          }
        }
      >
        Sign In
      </MenuItem>
      <MenuItem
        href="signup"
        onClick={[Function]}
        style={
          Object {
            "container": "menuItem__container",
          }
        }
      >
        Sign Up
      </MenuItem>
    </div>
  </div>
</header>
`;
